### PR TITLE
Space65 Updates

### PR DESCRIPTION
**Pull request by mechmerlin**
## Description
@mnpqraven's recent PR has coerced me into getting this in sooner rather than later =P 
- Implement Caps Lock LED support
- Add reset sequence doc
- General clean up here and there. 

## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [x] Enhancement/Optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by EskiMojo14** [comment]
Very minor detail, but could info.json be updated to show the correct proportions for the split space? The Space65 only supports 2.25-1.25-2.75 [shown here](https://geekhack.org/index.php?topic=98768.msg2699096#msg2699096) (and in the OP), whereas info.json has 2.75-1.25-2.25.

Didn't consider it worth opening a PR for, but if there's already efforts to clean it up I think it'd be nice. ☺️

**Comment by mechmerlin** [comment]
> Very minor detail, but could info.json be updated to show the correct proportions for the split space? The Space65 only supports 2.25-1.25-2.75 [shown here](https://geekhack.org/index.php?topic=98768.msg2699096#msg2699096) (and in the OP), whereas info.json has 2.75-1.25-2.25.
> 
> Didn't consider it worth opening a PR for, but if there's already efforts to clean it up I think it'd be nice. ☺️

No problem!

Just for anyone else going thru this. The size of the keys does not matter at all. This is purely a aesthetic change to match the space65 more closely. 

For example, the space bar is the fourth key in that row. It doesn't matter what size it is as long as it remains the fourth key in that row, it will still be programmed correctly.

File: keyboards/gray_studio/space65/keymaps/default/keymap.c [hunk 1]
```diff
@@ -23,10 +23,10 @@ enum custom_keycodes {
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 [0] = LAYOUT( \
-        KC_ESC,  KC_1,    KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_GRV, KC_INS,  \
-        KC_TAB,  KC_Q,    KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC, KC_DEL,          \
+        KC_ESC,  KC_1,    KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_DEL, \
+        KC_TAB,  KC_Q,    KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_INS,          \
         KC_CAPS, KC_A,    KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT, KC_ENT,  KC_PGUP,                  \
-        KC_LSFT, KC_NUBS, KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,  KC_PGDN,          \
+        KC_LSFT, KC_LSFT, KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_PGDN,         \
```
Comment (vomindoraan) [inline keyboards/gray_studio/space65/keymaps/default/keymap.c]: Just wondering, what's the reason for this change? Unless the regular 2.25u LShift uses the right pad instead of the left, leaving it as `KC_NUBS` should be fine, no?

File: keyboards/gray_studio/space65/space65.c [hunk 2]
```diff
@@ -38,6 +38,11 @@ bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
 
 void led_set_kb(uint8_t usb_led) {
 	// put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+  if (usb_led & (1 << USB_LED_CAPS_LOCK)) {
```
Comment (vomindoraan) [inline keyboards/gray_studio/space65/space65.c]:
```suggestion
  if (IS_LED_ON(usb_led, USB_LED_CAPS_LOCK)) {
```

Comment (vomindoraan) [reply keyboards/gray_studio/space65/space65.c]: You can also do
```suggestion
  (IS_LED_ON(usb_led, USB_LED_CAPS_LOCK) ? writePinHigh : writePinLow)(E6);
```
if you like that better.

File: keyboards/gray_studio/space65/readme.md [hunk 3]
```diff
@@ -1,11 +1,13 @@
 # Gray Studio Space65
 
-A 65% keyboard with RGB underglow, backlighting and USB C, whose design was inspired by the Voyager I space probe and Apple II home computer.
+A 65% keyboard with RGB underglow, backlighting, and USB C. Its design was inspired by the Voyager I space probe and Apple II home computer.
 
 Keyboard Maintainer: [MechMerlin](https://github.com/mechmerlin)  
 Hardware Supported: Gray Studio Space65  
 Hardware Availability: [Geekhack Group Buy](https://geekhack.org/index.php?topic=98768.0)
 
+**Reset Sequence:** Using this firmware sets `BOOTLOADER_ENABLE` to `lite`. While plugging in, hold the top left key, commonly programmed as `Esc` to put your board into bootloader mode.
```
Comment (vomindoraan) [inline keyboards/gray_studio/space65/readme.md]:
```suggestion
**Reset Sequence:** By default, `BOOTMAGIC_ENABLE` is set to `lite` on this board. Hold the top left key (commonly programmed as Esc) while plugging in to put the board into bootloader mode.
```

File: keyboards/gray_studio/space65/space65.h [hunk 4]
```diff
@@ -17,8 +17,6 @@
 
 #include "quantum.h"
 
-#define XXX KC_NO
```
Comment (vomindoraan) [inline keyboards/gray_studio/space65/space65.h]: Why don't you like `XXX`, again? I think it leads to better visual alignment.

Comment (mnpqraven) [reply keyboards/gray_studio/space65/space65.h]: I remember noroadsleft in a PR of mine [saying](https://github.com/qmk/qmk_firmware/pull/4704#pullrequestreview-187576269) about `KC_NO` and `KC_TRNS` being already defined in the quantum code and therefore don't need to be defined again for the sake of uniformity

Comment (vomindoraan) [reply keyboards/gray_studio/space65/space65.h]: Not the same thing since this is a board header rather than a keymap file. `XXX` is not a standard alias (yet).

Comment (vomindoraan) [reply keyboards/gray_studio/space65/space65.h]: We should probably standardize this at some point. At the moment, around 35 boards define `XXX` in their headers. Some boards use `xxx` and `_x_` instead.

File: keyboards/gray_studio/space65/keymaps/default/keymap.c (hunk 1 above)
Comment (mechmerlin) [reply keyboards/gray_studio/space65/keymaps/default/keymap.c]: It's done to match what the most common configuration of this board is.

File: keyboards/gray_studio/space65/readme.md (hunk 3 above)
Comment (mechmerlin) [reply keyboards/gray_studio/space65/readme.md]: I disagree with that statement. It has not been set to lite by default. I explicitly set it to lite.

File: keyboards/gray_studio/space65/space65.h (hunk 4 above)
Comment (mechmerlin) [reply keyboards/gray_studio/space65/space65.h]: I explicitly set it to KC_NO.

File: keyboards/gray_studio/space65/keymaps/default/keymap.c [hunk 5]
```diff
@@ -23,10 +23,10 @@ enum custom_keycodes {
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 [0] = LAYOUT( \
-        KC_ESC,  KC_1,    KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_GRV, KC_INS,  \
-        KC_TAB,  KC_Q,    KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC, KC_DEL,          \
+        KC_ESC,  KC_1,    KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_DEL, \
```
[the remark this answers is not in the document]
Comment (mechmerlin) [reply keyboards/gray_studio/space65/keymaps/default/keymap.c]: This is the most common configuration for a board like this.

File: keyboards/gray_studio/space65/space65.c (hunk 2 above)
Comment (vomindoraan) [reply keyboards/gray_studio/space65/space65.c]: Yeah, I agree. Just change it to use `IS_LED_ON` then.

File: keyboards/gray_studio/space65/readme.md (hunk 3 above)
Comment (vomindoraan) [reply keyboards/gray_studio/space65/readme.md]: That's what the sentence says, no? Note the “on this board” part. Anyway, I changed it because I thought “Using this firmware” sounded weird. Makes it sound like using _QMK_ sets `BOOTMAGIC_ENABLE` to `lite` 🤷‍♀️ Dunno how you could phrase that differently without it sounding awkward.

Comment (vomindoraan) [inline keyboards/gray_studio/space65/readme.md]:
```suggestion
**Reset Sequence:** `BOOTMAGIC_ENABLE` is set to [`lite`](/docs/feature_bootmagic.md#bootmagic-lite) on this board. Hold the top left key (commonly programmed as Esc) while plugging in to put your board into bootloader mode.
```
@mechmerlin Better?

Comment (mechmerlin) [reply keyboards/gray_studio/space65/readme.md]: I still disagree because what if you have kbdfans firmware? It's going to be space+b there.

File: keyboards/gray_studio/space65/space65.h (hunk 4 above)
Comment (noroadsleft) [reply keyboards/gray_studio/space65/space65.h]: Yeah, my comment was pertaining to keymaps rather than header files. I don't mind `XXXXXXX` or `_______` in keymaps; in fact, I use them myself.

For header files though, I've gotten to the point that I prefer using `KC_NO` directly, rather than `XXX`, `_x_`, or any of the other common variants. If we're going to standardize something for that usage, I'd be okay with standardizing on `XXX`, as I prefer that to any of the other variants I've seen, but I avoid it specifically because it's *not* standard.

File: keyboards/gray_studio/space65/space65.c (hunk 2 above)
Comment (pelrun) [reply keyboards/gray_studio/space65/space65.c]: Yikes, I've never considered using the ternary operator to select between function pointers. I'm pretty sure I don't like it :)

**Comment by drashna** [comment]
Awesome, thanks!

File: keyboards/gray_studio/space65/space65.c (hunk 2 above)
Comment (vomindoraan) [reply keyboards/gray_studio/space65/space65.c]: It's an idiom I came across in some embedded projects. It's like `++i %= len` in C++: at first you're like wtf, but then as you see it more and more you get used to it and it becomes fairly readable. But yes, it's not very noob friendly, and I wouldn't quite call it kernel quality code either.
